apps/server/src/shared/domain/service/permission.service.ts

```diff
@@ -53,10 +61,95 @@ export class PermissionService {
 		return hasPermissions;
 	}
 
-	checkUserHasAllSchoolPermissions(user: User, requiredPermissions: string[]): void {
-		const hasPermission = this.hasUserAllSchoolPermissions(user, requiredPermissions);
+	/**
+	 * Determines whether a user has all the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @throws UnauthorizedException
+	 */
+	checkUserHasAllPermissions(user: User, requiredPermissions: string[]): void {
+		const hasPermission = this.hasUserAllPermissions(user, requiredPermissions);
+		if (hasPermission !== true) {
+			throw new UnauthorizedException();
+		}
+	}
+
+	/**
+	 * Determines whether a user has at least one of the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @returns boolean
+	 */
+	hasUserOneOfPermissions(user: User, requiredPermissions: string[]): boolean {
+		if (!Array.isArray(requiredPermissions) || requiredPermissions.length === 0) {
+			return false;
+		}
+		const permissions = this.resolvePermissions(user);
+		const hasPermission = requiredPermissions.some((p) => permissions.includes(p));
+		return hasPermission;
+	}
+
+	/**
+	 * Determines whether a user has at least one of the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @throws UnauthorizedException
+	 */
+	checkUserHasOneOfPermissions(user: User, requiredPermissions: string[]): void {
+		const hasPermission = this.hasUserOneOfPermissions(user, requiredPermissions);
 		if (hasPermission !== true) {
 			throw new UnauthorizedException();
 		}
 	}
+
+	/**
+	 * Determines whether a user has access to the specified entity.
+	 * @example ```
+	 * const user = new User({id: 1})
+	 * const entity = new Course({id:2, creator: user})
+	 * const props = ['creator']
+	 * ```
+	 * @param user a user
+	 * @param entity An entity to access
+	 * @param props Array of properties in the entity the user is associated with
+	 * @returns
+	 */
+	hasUserAccessToEntity(user: User, entity: IEntity, props: string[]) {
+		const res = props.some((prop) => {
+			if (entity[prop] instanceof Collection) {
+				return (entity[prop] as Collection<IEntity, unknown>).contains(user);
+			}
+			if (entity[prop] instanceof User) {
+				return entity[prop] === user;
+			}
+			return entity[prop] === user.id;
+		});
+
+		return res;
+	}
+
+	/**
+	 * Determines whether a user has access to the specified entity with same scope.
+	 * @param user
+	 * @param entity
+	 * @returns boolean
+	 */
+	isOnSameSchool(user: User, entity: IEntityWithSchool) {
```

Comment:
On all other methods we use the name pattern has/check/is User OtherStruff. Then we should also do it on this place. Like isUserOnSameSchool.

But each of the checks need the user at first param. Maybe we can also remove User from each method.

hasAccressToEntity
checkHasOneOfPermissions
hasOneOfPermissions